sap-vcap-services/src/test/java/com/sap/cloud/environment/servicebinding/SapVcapServicesServiceBindingAccessorTest.java

```diff
@@ -128,53 +151,101 @@ void brokenBindingIsIgnored()
         assertContainsXsuaaBinding1(serviceBindings);
     }
 
+    @Test
+    void brokenBindingIsIgnoredWhenFileBased()
+    {
+        final String vcapServicesFilePath =
+            TestResource
+                .getPathAsString(SapVcapServicesServiceBindingAccessorTest.class, "VcapServicesWithBrokenBinding.json");
+
+        final SapVcapServicesServiceBindingAccessor sut =
+            new SapVcapServicesServiceBindingAccessor(
+                key -> "VCAP_SERVICES_FILE_PATH".equals(key) ? vcapServicesFilePath : null);
+
+        final List<ServiceBinding> serviceBindings = sut.getServiceBindings();
+
+        assertThat(serviceBindings.size()).isEqualTo(1);
+
+        assertContainsXsuaaBinding1(serviceBindings);
+    }
+
     @Test
     void resultIsNotCached()
     {
         final String vcapServices =
             TestResource.read(SapVcapServicesServiceBindingAccessorTest.class, "FullVcapServices.json");
-        final CountingEnvironmentVariableReader environmentVariableReader =
-            new CountingEnvironmentVariableReader("VCAP_SERVICES", vcapServices);
+        final CountingEnvironmentVariableReader environmentVariableReader = new CountingEnvironmentVariableReader();
+        environmentVariableReader.addExpectedKeyAndValue("VCAP_SERVICES", vcapServices);
+        environmentVariableReader.addExpectedKeyAndValue("VCAP_SERVICES_FILE_PATH", null);
+
+        final SapVcapServicesServiceBindingAccessor sut =
+            new SapVcapServicesServiceBindingAccessor(environmentVariableReader);
+
+        // first invocation
+        assertThat(sut.getServiceBindings().size()).isEqualTo(3);
+        assertThat(environmentVariableReader.getInvocations("VCAP_SERVICES")).isEqualTo(1);
+        assertThat(environmentVariableReader.getInvocations("VCAP_SERVICES_FILE_PATH")).isEqualTo(1);
+
+        // second invocation
+        assertThat(sut.getServiceBindings().size()).isEqualTo(3);
+        assertThat(environmentVariableReader.getInvocations("VCAP_SERVICES")).isEqualTo(2);
```

Comment:
for consistency, why not:

```suggestion
        assertThat(environmentVariableReader.getInvocations("VCAP_SERVICES")).isEqualTo(2);
        assertThat(environmentVariableReader.getInvocations("VCAP_SERVICES_FILE_PATH")).isEqualTo(2);
```

Reply:
I see you added this condition in the test case below, but then there is no point in having both tests, as they are basically identical, right?

Reply:
The new one `resultIsNotCachedWhenFileBased` sets the VCAP_SERVICES to `null` and checks that `VCAP_SERVICES_FILE_PATH` is really accessed, but a check that the file has been read again is not easily possible. I tried to define the read of the file similar to `public static final Function<String, String> DEFAULT_ENVIRONMENT_VARIABLE_READER = System::getenv;` so that I can override it in the test but the code readability suffered. If you like I can change it so that we really test that the read was executed. At the moment it just check that the VCAP_SERVICES_FILE_PATH has been accessed and the value of the bindings is fine.